hera_cal/frf.py

```diff
@@ -230,21 +246,28 @@ def timeavg_data(self, t_avg, rephase=False, verbose=True):
         t_avg = Navg * self.dtime
 
         if verbose:
-            print("The t_avg provided of {:.1f} has been shifted to {:.1f} to make Navg = {:d}".format(
+            print("The t_avg provided of {:.3f} has been shifted to {:.3f} to make Navg = {:d}".format(
                 old_t_avg, t_avg, Navg))
 
         # setup lists
-        avg_data = odict()
-        avg_flags = odict()
-        avg_nsamples = odict()
+        avg_data = DataContainer({})
+        avg_flags = DataContainer({})
+        avg_nsamples = DataContainer({})
+
+        # setup averaging quantities
+        if data is None:
```

Comment:
Again, same comment as before. This feels like a weird hijacking of the object rather than a natural use of the underlying functions.

Reply:
I don't think its going against the philosophy of the object. The point is to be able to build upon the objects inherent capabilities: `VisClean` can take `self.data` and from it create `self.clean_model` and `self.clean_resid` and `self.clean_data`. One can choose to `timeavg` any one of those by simply feeding them into the function as `timeavg(data=VC.clean_resid)`, for example. Now one thing you may be thinking is, well we could do `data=['data', 'model', 'resid'...]` as a predefined list of choices, which it then knows to pick out `self.data`, `self.model`, `self.resid` etc respectively, but that seems a little unnecessarily rigid. I guess what I can say is, I have already found it beneficial to work with the object in this way in my own analyses, rather than having it hardcoded.